flocker/volume/filesystems/zfs.py

```diff
@@ -67,26 +71,100 @@ def connectionLost(self, reason):
         del self._result
 
 
-def zfs_command(reactor, arguments):
+class _AsyncLZC(object):
+    def __init__(self, reactor):
+        self._reactor = reactor
+        self._cache = {}
+
+    def callDeferred(self, func, *args, **kwargs):
+        return deferToThreadPool(self._reactor, self._reactor.getThreadPool(),
+                                 func, *args, **kwargs)
+
+    def __getattr__(self, name):
+        try:
+            return self._cache[name]
+        except KeyError:
+            func = getattr(libzfs_core, name)
+
+            @wraps(func)
+            def _async_wrapper(*args, **kwargs):
+                return self.callDeferred(func, *args, **kwargs)
+
+            self._cache[name] = _async_wrapper
+            return self._cache[name]
+
+
+class _FakeAsyncLZC(object):
+    def __init__(self):
+        self._cache = {}
+
+    def callDeferred(self, func, *args, **kwargs):
+        return maybeDeferred(func, *args, **kwargs)
+
+    def __getattr__(self, name):
+        try:
+            return self._cache[name]
+        except KeyError:
+            func = getattr(libzfs_core, name)
+
+            @wraps(func)
+            def _async_wrapper(*args, **kwargs):
+                return maybeDeferred(func, *args, **kwargs)
+
+            self._cache[name] = _async_wrapper
+            return self._cache[name]
+
+
+_reactor_to_alzc = {}
+
+
+def _async_lzc(reactor):
```

Comment:
The missing documentation would help here as well :-(
The main purpose of this function is to associate an `AsyncLZC` instance with a given reactor and cache it.